### DISCUSSION TITLE
Search extensibility library - How to use Async / Promise with Handlebars?

### DISCUSSION THREAD

**Discussion by IrCheema** [microsoft-search/pnp-modern-search#3465] [Q&A]
Hi @wobba ,
I want to use Chart.Js in my Search extensibility Library to show LineChart for every result . I am stuck at a point where i want to use  data from a promise/ async call in a Handlebar helper function so it can be used via props. It seems handlebars can not be used with async calls. Is there any way to use async function. For example adding an EventListne like onMouseover to update the hover element after rendering the Handlebars template. The solution is working but got a huge performance hit due to "async: flase". Any suggestion /help  will be appreciated.
![image](https://github.com/microsoft-search/pnp-modern-search/assets/22212050/87d8baad-5b5b-4f38-b02a-1bf591231093)
![image](https://github.com/microsoft-search/pnp-modern-search/assets/22212050/a91f7f33-8345-44bc-afe4-a38e6fe47e59)
![image](https://github.com/microsoft-search/pnp-modern-search/assets/22212050/8128039b-cd72-47dd-b62a-760b3553c190)
![image](https://github.com/microsoft-search/pnp-modern-search/assets/22212050/f69992d7-95ed-41f9-8c7d-8527f4c58009)

**Comment by wobba** [comment]
You have a custom web component right? You can do any async calling you in there... same as in SPFx. Make the render function async as an example. And why use jQuery for ajax when in SPFx? :)

**Reply by IrCheema** [reply]
@wooba thanks for taking time to answer. jQuery is an old pal that gives me option to switch between async and sync calls. While I tried your suggestion and it works to some extent. I don't have much experience with React and SPFX so please bare with me. I have  moved the Chart control and logic  to my Component and change the render function and ajax call to async. But I think the control is rendered before the ajax call gets completed.
![image](https://github.com/microsoft-search/pnp-modern-search/assets/22212050/d9de7af7-8de8-40ed-a7ba-2c91b7c0b7e7)
![image](https://github.com/microsoft-search/pnp-modern-search/assets/22212050/35fb031a-3bcf-42cc-876a-6e365b64f2b1)
![image](https://github.com/microsoft-search/pnp-modern-search/assets/22212050/de410849-87b8-46b9-b890-fb746df7be24)
![image](https://github.com/microsoft-search/pnp-modern-search/assets/22212050/de002264-17aa-46ce-bcf3-bbf810a33f2e)

**Reply by wobba** [reply]
General SPFx questions can be asked at https://github.com/SharePoint/sp-dev-docs and it's all about figuring out React lifecycle and Typescript.
